src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -306,13 +314,67 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                         End If
                     Loop
```

Comment:
>Loop [](start = 20, length = 4)

This preexisting condition, but it looks like containers can take advantage of similar optimizations around definitions as we are doing before entering ```If Not (t1IsDefinition AndAlso t2IsDefinition) Then```.